meson.build

```diff
@@ -0,0 +1,1629 @@
+endif
+
+vaapi_x11 = {
+    'name': 'vaapi-x11',
+    'deps': dependency('libva-x11', version: '>= 1.1.0', required: get_option('vaapi-x11')),
+}
+vaapi_x11 += {'use': vaapi['use'] and x11['use'] and vaapi_x11['deps'].found()}
+if vaapi_x11['use'] and get_option('gpl').enabled()
+    dependencies += vaapi_x11['deps']
+    sources += 'video/out/vo_vaapi.c'
+endif
+
+vaapi_wayland = {
+    'name': 'vaapi-wayland',
+    'deps': dependency('libva-wayland', version: '>= 1.1.0', required: get_option('vaapi-wayland')),
+}
+vaapi_wayland += {'use': vaapi['use'] and egl_wayland['use'] and vaapi_wayland['deps'].found()}
+
+vaapi_drm = {
+    'name': 'vaapi-drm',
+    'deps': dependency('libva-drm', version: '>= 1.1.0', required: get_option('vaapi-drm')),
+}
+vaapi_drm += {'use': vaapi['use'] and egl_drm['use'] and vaapi_drm['deps'].found()}
+
+vaapi_x_egl = {
+    'name': 'vaapi-x-egl',
+    'use': vaapi_x11['use'] and egl_x11['use'],
+}
+
+vaapi_egl = {
+    'name': 'vaapi-egl',
+    'use': vaapi_x_egl['use'] or vaapi_wayland['use'] or vaapi_drm['use'],
+}
+if vaapi_egl['use']
+    dependencies += [vaapi_wayland['deps'], vaapi_drm['deps']]
+    sources += 'video/out/hwdec/hwdec_vaapi_gl.c'
+endif
+
+vaapi_vulkan = {
+    'name': 'vaapi-vulkan',
+    'use': vaapi['use'] and libplacebo.found() and vulkan.found(),
+}
+if vaapi_vulkan['use']
+    sources += 'video/out/hwdec/hwdec_vaapi_vk.c'
+endif
+
+if vaapi_egl['use'] or vaapi_vulkan['use']
+    sources += 'video/out/hwdec/hwdec_vaapi.c'
+endif
+
+ffnvcodec = dependency('ffnvcodec', version: '>= 8.2.15.7', required: get_option('cuda-hwaccel'))
+if ffnvcodec.found()
+    dependencies += ffnvcodec
+    sources += 'video/cuda.c'
+endif
+
+if ffnvcodec.found() and not get_option('cuda-interop').disabled()
+    sources += 'video/out/hwdec/hwdec_cuda.c'
+endif
+
+if ffnvcodec.found() and not get_option('cuda-interop').disabled() and gl['use']
+    sources += 'video/out/hwdec/hwdec_cuda_gl.c'
+endif
+
+if ffnvcodec.found() and vulkan.found() and libplacebo.found() and not get_option('cuda-interop').disabled()
+    sources += 'video/out/hwdec/hwdec_cuda_vk.c'
+endif
+
+
+# extra options
+if get_option('tests').enabled()
+    sources += ['test/champ.c',
+                'test/gl_video.c',
+                'test/img_format.c',
+                'test/json.c',
+                'test/linked_list.c',
+                'test/paths.c',
+                'test/scale_sws.c',
+                'test/scale_test.c',
+                'test/tests.c']
+endif
+
+if get_option('tests').enabled() and zimg.found()
+    sources += ['test/repack.c',
+                'test/scale_zimg.c']
+endif
+
+manpage = 'DOCS/man/mpv.rst'
+
+rst2man = find_program('rst2man', required: get_option('manpage-build'))
+if rst2man.found()
+    out_dir = join_paths(meson.build_root(), 'DOCS', 'man')
+    run_command('mkdir', '-p', out_dir)
+    run_command('rst2man', '--strip-elements-with-class=contents', manpage, join_paths(out_dir, 'mpv.1'))
```

Comment:
OK, again, this really should be a custom_target so that it gets rebuilt if it changes after a git pull. And so that you're not using mkdir and stuff on Windows. :/

Also a custom_target can be directly installed without install_data -- just add the kwargs `install: true, install_dir: join_paths(mandir, 'man1')`

Reply:
Agree about custom_target(). Moreover, you can just use `install_dir: '@mandir@'`